### PR TITLE
Bump versions for Iron Patch 3

### PR DESCRIPTION
**Pull request by Yadunund**
TODO:
- [x] Check if rosbag2 needs to be bumped again: https://github.com/ros2/rosbag2/pull/1487#issuecomment-1815832034
- [x] CI job with new ros2.repos file after PR is approved
- [x] Packaging job after CI passes
- [ ] Tag source release

### CODE REVIEW

File: ros2.repos [hunk 1]
```diff
@@ -334,7 +334,7 @@ repositories:
   ros2/rosbag2:
     type: git
     url: https://github.com/ros2/rosbag2.git
-    version: 0.22.3
+    version: 0.22.4
```
Comment (clalancette) [inline ros2.repos:337]: I think we are going to need to do a 0.22.5 release to fix up the tag here.  See https://github.com/ros2/rosbag2/pull/1487#issuecomment-1816407337

**Review by clalancette — CHANGES_REQUESTED** [review]
This looks good with the one exception of rosbag2.

**Comment by Yadunund** [comment]
CI job created using https://raw.githubusercontent.com/ros2/ros2/yadu/iron-patch-3/ros2.repos

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19818/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14318/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20539/)
* RHEL-9 [Build Status](https://ci.ros2.org/view/All/job/ci_linux-rhel/515/)

Will run packaging jobs after CI passes.

File: ros2.repos (hunk 1 above)
Comment (Yadunund) [reply ros2.repos]: https://github.com/ros2/ros2/pull/1497/commits/2797436bc78aae1ad1d78d525ee4a30e380949d7

**Comment by Yadunund** [comment]
Packaging:
* Linux [Build Status](https://ci.ros2.org/view/packaging/job/packaging_linux/3237/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/view/packaging/job/packaging_linux-aarch64/2601/)
* Windows [Build Status](https://ci.ros2.org/view/packaging/job/packaging_windows/3058/)
* Windows Debug [Build Status](https://ci.ros2.org/view/packaging/job/packaging_windows_debug/2017/)
* RHEL-9 [Build Status](https://ci.ros2.org/view/packaging/job/packaging_linux-rhel/1738/)

**Comment by clalancette** [comment]
The packaging jobs look good.

The CI jobs are failing for the same reason as the Humble jobs, though I actually don't fully understand what is happening.  With our new RTI license, that should be working.  We'll have to take a closer look at that, but I think we can proceed with those things failing.

**Review by clalancette — APPROVED** [review]

**Pull request merged**